test/integration/node-specific/abstract_cursor.test.ts

```diff
@@ -473,39 +477,59 @@ describe('class AbstractCursor', function () {
     });
 
     describe('when the cursor refreshes the timeout for killCursors', function () {
-      it(
-        'the provided timeoutContext is not modified',
-        {
-          requires: {
-            mongodb: '>=4.4'
-          }
-        },
-        async function () {
-          await client.db('admin').command({
+      let uri: string;
+
+      before(function () {
+        uri = this.configuration.url({ useMultipleMongoses: false });
+      });
+
+      beforeEach(async function () {
+        commands.length = 0;
+        await configureFailPoint(
+          this.configuration,
+          {
             configureFailPoint: 'failCommand',
             mode: { times: 1 },
             data: {
               failCommands: ['getMore'],
               blockConnection: true,
               blockTimeMS: 5000
             }
-          } as FailPoint);
+          },
+          uri
+        );
+      });
+
+      afterEach(async function () {
+        await clearFailPoint(this.configuration, uri);
+      });
 
+      it(
+        'the provided timeoutContext is not modified',
+        {
+          requires: {
+            mongodb: '>=4.4',
+            topology: '!load-balanced'
+          }
+        },
+        async function () {
           const cursor = collection.find(
             {},
             {
               timeoutContext: context,
-              timeoutMS: 1000,
+              timeoutMS: 150,
               timeoutMode: CursorTimeoutMode.LIFETIME,
               batchSize: 1
             }
           );
 
+          const refresh = sinon.spy(context, 'refresh');
```

Comment:
Checking that `remainingTimeoutMS < 0` was flaky.  I don't particularly like this approach, because it relies on testing the current implementation, but this does fix the flakiness.